src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/ConfigurationLiveTests.cs

```diff
@@ -456,13 +455,12 @@ public async Task Get()
             {
                 await service.SetAsync(testSettingNoLabel);
                 // Test
-                Response<ConfigurationSetting> response = await service.GetAsync(key: testSettingNoLabel.Key, options: default, CancellationToken.None);
+                ResponseTask<ConfigurationSetting> task = service.GetAsync(key: testSettingNoLabel.Key, options: default, CancellationToken.None);
+                await task;
+                Assert.True(task.Response.TryGetHeader("ETag", out string etagHeader));
 
-                Assert.True(response.TryGetHeader("ETag", out string etagHeader));
-
-                ConfigurationSetting setting = response.Result;
+                ConfigurationSetting setting = task.Result;
```

Comment:
I do wonder whether you'd be better served by just having the operation return a tuple of the response and the setting, e.g.
```C#
var (response, setting) = await service.GetAsync(key: testSettingNoLabel.Key, options: default, CancellationToken.None);
Assert.True(response.TryGetHeader("ETag", out string etagHeader));
Assert.AreEqual(testSettingNoLabel, setting);
```
I'm not seeing much advantage to having this custom task-like type in this case.  Plus, the boilerplate required to create it is a lot more than the boilerplate around a tuple, whether just using the language support for tuple (in which case there's basically no boilerplate) or a custom type (in which case it could just be a simple POCO class or struct).

While the language supports pattern-based returns, my preference is for that to be an implementation detail inside of a library.  For public APIs, my preference is for everything to just return `{Value}Task{<T>}`.

Reply:
The current pattern (before this PR) is returning a strongly typed tuple with a deconstruct method and with a cast to T. The problem is that people don't seem to like it. The main two complaints are:
1. The signature of the service call methods is very complicated (btw, with real tuples it would be even more complicated).
2. People don't realize that they can cast nor deconstruct.

Reply:
I hear you, and some of this is eye-of-the-beholder... from my perspective, though, this is a step backwards, harder to both produce and consume.  My $.02.

Reply:
Why do you think it makes it harder to consume? We only care about being to easily consume the T. Consuming headers is an advanced scenario.

Reply:
> Why do you think it makes it harder to consume?

Consuming isn't just about the direct await, but also about when consuming and producing, e.g. you write an intermediate function that awaits X, does something, and then returns the result of X.  That's much harder with this approach.

Reply:
Why is it harder than with Task?, i.e. 
```c#
await X;
DoSomething();
return x.Result;
```

Reply:
```C#
private readonly SemaphoreSlim _throttle = new SemaphoreSlim(1, 1);

public static async ResultTask<T> OuterAsync()
{
    await _throttle.WaitAsync();
    try
    {
        return await InnerAsync(); // this will not compile... what do I need to write instead?
    }
    finally
    {
        _throttle.Release();
    }
}
```

Reply:
In contrast, this works just fine:
```C#
private readonly SemaphoreSlim _throttle = new SemaphoreSlim(1, 1);

public async (T Result, AzureResponse Response) OuterAsync()
{
    await _throttle.WaitAsync();
    try
    {
        return await InnerAsync();
    }
    finally
    {
        _throttle.Release();
    }
}
```

Reply:
Yeah, you would have to write:
```c#
private readonly SemaphoreSlim _throttle = new SemaphoreSlim(1, 1);

public async ResponseTask<T> OuterAsync()
{
    await _throttle.WaitAsync();
    try
    {
        var ret = InnerAsync();
        await ret;
        return ret;
    }
    finally
    {
        _throttle.Release();
    }
}
```

Reply:
That doesn't work.  `ret` here is a `ResponseTask<T>`; the `return ret` will fail to compile, because you need to return a `T`.  And if you return `ret.Task`, then you won't be propagating the ret.Response.

Reply:
OR:
```C#
private readonly SemaphoreSlim _throttle = new SemaphoreSlim(1, 1);

public async ResponseTask<T> OuterAsync()
{
    await _throttle.WaitAsync();
    try
    {
        return InnerAsync().OurWait(); // this is just helped doing the above
    }
    finally
    {
        _throttle.Release();
    }
}
```

Reply:
> OR

Same problem.

Reply:
But I get your point now.

Reply:
This seems to work:
```c#
        async ResponseTask<ConfigurationSetting> Foo(ConfigurationClient client)
        {
            await Task.Delay(100);
            await client.GetAsync("").Await();
        }

        public static async ResponseTask<T> Await<T>(this ResponseTask<T> task)
        {
            var result = await task;
            return await FromResult(result, task.Response);
        }
```

Reply:
> This seems to work

I don't see how it can.  `await FromResult(result, task.Response)` will return a `T`; the `ResponseTask<T>` returned from `Await<T>` will not contain `task.Response`.

Reply:
How do we make it work for ValueTask?

Reply:
`ValueTask<T>` doesn't carry secondary state on it.  There's no additional information that's needed to construct one.